README.md

```diff
@@ -1,4 +1,371 @@
+In Azure CLI, we allow user to set the min or max api versions of a specific command groups or command or parameters. We can configure in readme.az.md so the generate code can work in that way too.  
+For example:
+```
+cli:
+  cli-directive:
+    - where:
+        group: groupCondition
+        op: opCondition
+        param: paramCondition
+      min-api: 2019-01-01
+      max-api: 2020-12-01
+```
+* Note: you don't need to specify both the min-api and max-api. and the group, op, param conditions are not all necessary either.   
+
+### **move command groups/command layer**  
+Before we talk about move command groups and command layer. we need to have some basic ideas about what command group name comes from and what command name comes from.  
+
+As we probably know that in Swagger the operationId are usually in the format of **A_B** where **A** is resource name in the format of plural and **B** is the action name you want to perform on that resource for example create, update, get, start, stop, delete etc.   
+
+In CLI code generation, we view **A** as group name, **B** as the command name and the CLI command of operationId **A_B** would be like `az <extension-name> A B`.  
+    
+In Azure CLI it's quite common that we want to move the same functional command into the same command group. For example:   
+```
+directive:
+    ## remove a sub group share
+    - where:
+          group: datashare share
+      set:
+          group: datashare
+
+    ## add a sub group consumer
+    - where:
+          group: datashare trigger
+      set:
+          group: datashare consumer trigger
+
+    ## change the group and name of a command
+    - where:
+          command: datafactory integration-runtime create-linked-integration-runtime
+      set:
+          command: datafactory integration-runtime linked-integration-runtime create
+
+
+```
+See [how to add or remove subgroups](https://github.com/Azure/autorest.az/blob/master/doc/faq.md#how-to-addremove-subgroup) for more details.   
+
+### **rename/hide command groups, commands, parameters**  
+We provide the ability for user to rename or hide command groups or commands or parameters. For example:
+```
+cli:
+  cli-directive:
+    ## rename a parameter 
+    - where:
+        group: groupCondition
+        op: opCondition
+        param: paramCondition
+      name: new_op_name
+
+    ## hide an operation
+    - where:
+        group: groupCondition
+        op: opCondition
+      hide: true    
+```
+* Note: if a parameter has the flattened schema prefix in the name, then we can't rename it in this way, because in Autorest.Clicommon it doesn't have the flattened schema prefix. We can only add alias for this parameter in such case.
+
+
+### **Client Factory Customization**
+By default, the client factory should not expose the subscriptionId, baseUrl parameters to customer as well as authentication policy but we provide customization for these scenarios.
```

Comment:
for each of the advanced feature, is it ok to add a real RP as example, so people could check RP behavior for better understandign?

Reply:
Do you mean the generated code example? I have provide the readme.az.md configuration example for most of them.  And I was thinking maybe we can add a user scanrios folder and CLI guys can add some real user scenarios while onboarding RP.